clients/java/signalr/src/main/java/com/microsoft/signalr/HubException.java

```diff
@@ -3,16 +3,32 @@
 
 package com.microsoft.signalr;
 
+/**
+ * The exception thrown from a hub when an error occurs.
+ */
 public class HubException extends RuntimeException {
     private static final long serialVersionUID = -572019264269821519L;
 
+    /**
+     * Initializes a new instance of the {@link HubException} class.
+     */
     public HubException() {
     }
 
+    /**
+     * Initializes a new instance of the {@link HubException} class with a specified error message
+     * @param errorMessage The error message that explains the reason for the exception.
+     */
     public HubException(String errorMessage) {
         super(errorMessage);
     }
 
+    /**
+     * Initializes a new instance of the {@link HubException} class with a specified error message and a reference
+     * to the inner exception that is the cause of this exception.
```

Comment:
Suggest: ...a reference to the ~~inner~~ exception that is the cause of this exception.

Sounds like "inner exception" is a .NET term, and "cause" is the preferred term in Java. No need to mix in .NET terms here. For example, see [Javadocs for `Exception`](https://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html#Exception(java.lang.String,%20java.lang.Throwable))

Reply:
Feel free to just copy the Javadocs off the official ones for Exception if you think that's more appropriate :). Up to you.